src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -1,59 +1,65 @@
 package com.google.sps.servlets;
 
 import com.google.sps.data.UserData;
+import com.google.sps.data.FirebaseAuthentication;
+import com.google.sps.data.FirebaseSingletonApp;
 import com.google.gson.Gson;
-import java.util.List;
 import java.io.IOException;
-import java.util.ArrayList;
 import com.google.firebase.FirebaseApp;
-import com.google.firebase.FirebaseOptions;
 import com.google.firebase.auth.*;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
-import com.google.appengine.api.datastore.Query.SortDirection;
+import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /**
- * Servlet that will add a user with his address to the database when he sets it first.
- * Users can also edit their adrress. 
+ * Servlet that will add a user with his address to the database when he sets it
+ * first. Users can also edit their adrress.
  */
 @WebServlet("/user-data")
 public class UserDataServlet extends HttpServlet {
-  
+  private FirebaseAuthentication firebaseAuth;
+
+   @Override
+  public void init() throws ServletException {
+    // initialize firebase app 
+    firebaseAuth = new FirebaseAuthentication(FirebaseSingletonApp.getInstance());
+  }
+
+  public void setFirebaseAuth(FirebaseAuthentication firebaseAuth) {
+    this.firebaseAuth = firebaseAuth;
+  }
+
   // Get all info about current user.
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-  
-    // Initialize firebase app.
-    FirebaseApp defaultApp = initializeFirebaseApp(/* setOptions */ false);
     String uid = null;
     try {
-      // get the idToken from hidden input field
-      String idToken = request.getParameter("idToken");
-      FirebaseToken decodedToken = FirebaseAuth.getInstance().verifyIdToken(idToken);
-      uid = decodedToken.getUid();
+      // get the idToken from query string
+      uid = firebaseAuth.getUserIdFromIdToken(request.getParameter("idToken"));
     } catch (FirebaseAuthException e) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid idToken");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
       return;
     }
 
-    /* Return all info about the current user if he has been added to the database. */
+    /*
```

Comment:
Please stop fixing unrelated stuff in PRs :(

If it bothers you do a completely separate PR to fix the whitespace.